### PR TITLE
[skip ci] ceph-container-engine: allow override container_package_name and container_service_name

### PR DESCRIPTION
**Pull request by clwluvw**
Only include specific variables when they are undefined

Signed-off-by: Seena Fallah <seenafallah@gmail.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test centos-non_container-lvm_osds

File: roles/ceph-container-engine/tasks/pre_requisites/prerequisites.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@
   with_first_found:
     - "{{ ansible_facts['distribution'] }}-{{ ansible_facts['distribution_major_version'] }}.yml"
     - "{{ ansible_facts['os_family'] }}.yml"
+  when: container_package_name is undefined or container_service_name is undefined
```
Comment (guits) [inline roles/ceph-container-engine/tasks/pre_requisites/prerequisites.yml:7]:
```suggestion
  when: container_package_name is undefined and container_service_name is undefined
```

**Review by guits — CHANGES_REQUESTED** [review]

File: roles/ceph-container-engine/tasks/pre_requisites/prerequisites.yml (hunk 1 above)
Comment (guits) [reply roles/ceph-container-engine/tasks/pre_requisites/prerequisites.yml]: true. maybe adding a check in the ceph-validate role makes sense?

Comment (clwluvw) [reply roles/ceph-container-engine/tasks/pre_requisites/prerequisites.yml]: @guits Added. PTAL

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-lvm_osds

**Comment by guits** [comment]
maybe another option would be to set a fact before the task `include specific variables` like the following :

```yaml
- name: set_fact _container_package_name, _container_service_name
  set_fact:
    _container_package_name: "{{ container_package_name if container_package_name is defined | bool else omit }}"
    _container_service_name: "{{ container_service_name if container_service_name is defined | bool else omit }}" 

- name: include specific variables
  include_vars: "{{ item }}"
  with_first_found:
    - "{{ ansible_facts['distribution'] }}-{{ ansible_facts['distribution_major_version'] }}.yml"
    - "{{ ansible_facts['os_family'] }}.yml"
```

then in all files in `./roles/ceph-container-engine/vars/` we would modify variables like this:

```yaml
$ cat ./roles/ceph-container-engine/vars/Debian.yml
---
container_package_name: "{{ _container_package_name | default('docker-ce') }}"
container_service_name: "{{ _container_service_name | default('docker') }}"
```

It seems to work **but** :
- not thoroughly tested,
- that's adding a bit more complexity,
- I'm not sure if it can bring any side effects that wouldn't be caught by the CI

We might prefer to stick to something very basic.

@dsavineau thoughts?

**Comment by clwluvw** [comment]
I think the current implementation is simpler than this @guits

**Review by guits — APPROVED** [review]

**Comment by guits** [comment]
> I think the current implementation is simpler than this @guits

yes it is

**Comment by clwluvw** [comment]
jenkins test centos-non_container-collocation

**Comment by clwluvw** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
@clwluvw the two failures aren't related to your patch

**Comment by clwluvw** [comment]
> @clwluvw the two failures aren't related to your patch

@guits Can it be merged and backport without ci or should wait for ci to be fixed?

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-collocation

**Review by guits — APPROVED** [review]

**Pull request merged**